src/aliceVision/lensCorrectionProfile/lcp.cpp

```diff
@@ -0,0 +1,1031 @@
+        {
+            for (int i = 0; attr[i]; i += 2)
+            {
+                std::string key(attr[i]);
+                std::string value(attr[i + 1]);
+
+                LCPdata->_currText = value;
+                LCPdata->setCommonSettings(key);
+            }
+        }
+        else
+        {
+            LCPdata->setGetText();
+        }
+    }
+
+}  /* End of start handler */
+
+void XMLCALL LCPinfo::XmlEndHandlerCommonOnly(void* pLCPinfo, const char* el)
+{
+    LCPinfo* LCPdata = static_cast<LCPinfo*>(pLCPinfo);
+
+    std::string element(el);
+
+    if (LCPdata->_currReadingState == LCPReadingState::FillCommonAndCameraSettings)
+    {
+        if (!LCPdata->_currText.empty())
+        {
+            LCPdata->setCommonSettings(element);
+            LCPdata->_currText.clear();
+        }
+    }
+}  /* End of end handler */
+
+// LCPinfo class implementation
+
+LCPinfo::LCPinfo(const std::string& filename, bool fullParsing) :
+    _isSeqOpened(false),
+    _isCommonOK(false),
+    _isCamDataOK(false),
+    _inAlternateLensIDs(false),
+    _inAlternateLensNames(false),
+    _waitPerspModeldescription(false),
+    _getText(false),
+    _modelCount(0),
+    Author(""),
+    Make(""),
+    Model(""),
+    UniqueCameraModel(""),
+    CameraRawProfile(false),
+    LensInfo(""),
+    CameraPrettyName(""),
+    LensPrettyName(""),
+    ProfileName(""),
+    SensorFormatFactor(1.f),
+    ImageWidth(0),
+    ImageLength(0)
+{
+    load(filename, fullParsing);
+}
+
+void LCPinfo::load(const std::string& filename, bool fullParsing)
+{
+    if(fullParsing)
+    {
+        ALICEVISION_LOG_INFO("Load LCP file: \"" << filename << "\"");
+    }
+    else
+    {
+        ALICEVISION_LOG_TRACE("Load LCP header: \"" << filename << "\"");
+    }
+    XML_Parser parser = XML_ParserCreate(nullptr);
+
+    if (!parser)
+    {
+        throw std::runtime_error("Couldn't allocate memory for XML parser");
+    }
+
+    if (fullParsing)
+    {
+        XML_SetElementHandler(parser, XmlStartHandler, XmlEndHandler);
+    }
+    else
+    {
+        XML_SetElementHandler(parser, XmlStartHandlerCommonOnly, XmlEndHandlerCommonOnly);
+    }
+    
+    XML_SetCharacterDataHandler(parser, XmlTextHandler);
+    XML_SetUserData(parser, static_cast<void*>(this));
+
+    FILE* const pFile = fopen(filename.c_str(), "rb");
+
+    if (pFile) {
+        constexpr int BufferSize = 8192;
+        char buf[BufferSize];
+        bool done;
+
+        do {
+            int bytesRead = fread(buf, 1, BufferSize, pFile);
+            done = feof(pFile);
+
+            if (XML_Parse(parser, buf, bytesRead, done) == XML_STATUS_ERROR) {
+                XML_ParserFree(parser);
+                throw std::runtime_error("Invalid XML in LCP file");
+            }
+        } while (!done);
+
+        fclose(pFile);
+    }
+
+    XML_ParserFree(parser);
+}
+
+bool LCPinfo::search(settingsInfo& settings, LCPCorrectionMode mode, int& iLow, int& iHigh, float& weightLow)
```

Comment:
could it be refactored ? (same for the other methods marked as complex by CodeFactor)